src/containers/GuestInfoContainer.tsx

```diff
@@ -0,0 +1,170 @@
+      .required('Birthdate is required')
+  });
+
+  const [privacy, setPrivacy] = useState<boolean>(false);
+  const [error, setError] = useState<undefined | string>();
+
+  const methods = useForm<PersonalInfo>({
+    resolver: yupResolver(NewUserSchema),
+    defaultValues
+  });
+
+  const {
+    watch,
+    handleSubmit,
+    formState: { isSubmitting }
+  } = methods;
+
+  const onSubmit = useCallback(
+    async (values: PersonalInfo) => {
+      logger.info('submit user data', values);
+      try {
+        setError(undefined);
+
+        if (checkout === undefined) {
+          throw Error('Priced offer undefined');
+        }
+        if (!checkout.offerId) {
+          throw Error('Priced offer offerId not defined');
+        }
+        if (values.birthdate === null) {
+          throw Error('Birth date is not defined');
+        }
+
+        const formattedDate = convertToLocalTime(values.birthdate);
+        await axios
+          .request(
+            new PersonalInfoRequest(checkout.offerId, {
+              ...values,
+              birthdate: formattedDate
+            })
+          )
+          .catch(() => {
+            throw Error('Something went wrong');
+          });
+
+        dispatch({
+          type: 'SET_CHECKOUT',
+          payload: {
+            ...checkout,
+            personalInfo: { ...values, birthdate: formattedDate }
+          }
+        });
+
+        logger.info('Guest info sent successfully');
+        navigate('/checkout/' + checkout.offerId);
+      } catch (error) {
+        const message = (error as Error).message || 'Unknown error';
+        setError(message);
+      }
+    },
+    [checkout, setError]
+  );
+
+  return (
+    <FormProvider methods={methods} onSubmit={handleSubmit(onSubmit)}>
+      <Grid justifyContent="center" container spacing={3}>
+        <Grid item xs={12} md={6}>
+          <Typography textAlign="center" sx={{ m: 2 }}>
+            This Information will be strictly used by the hotel and win.so to confirm your
+            booking! It is mandatory in order for the hotel to reserve your room.
+          </Typography>
+          {error && <Alert severity="error">{error}</Alert>}
+          <Card sx={{ my: 2, p: 3 }}>
+            <Box
+              sx={{
+                display: 'grid',
+                columnGap: 2,
+                rowGap: 3,
+                gridTemplateColumns: { xs: 'repeat(1, 1fr)', sm: 'repeat(1, 1fr)' }
+              }}
+            >
+              <RHFTextField name="firstname" label="First Name" />
+              <RHFTextField name="lastname" label="Last Name" />
+              <RHFDatePicker name="birthdate" label="Birth date" />
+              <RHFTextField name="email" label="Email Address" />
+              <RHFTextField name="phone" label="Phone Number" />
+            </Box>
+            <Stack direction="row" alignItems="flex-start" sx={{ mt: 3 }}>
+              <Checkbox checked={privacy} onChange={() => setPrivacy(!privacy)} />
```

Comment:
Can users now not continue to the next page, without approving this privacy policy?

Reply:
did not get the question. submit button is available only if checked

Reply:
ok perfect